pkg/controller/statusmanager/status_manager.go

```diff
@@ -379,8 +380,37 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...operv
 			)
 		}
 
-		if oc.Spec.DefaultNetwork.Type == operv1.NetworkTypeOpenShiftSDN {
-			// OpenShiftSDN is removed in 4.17, so block the upgrade if we have OpenShiftSDN in the spec.
+		migrationInProgress := false
+		netCfg := &configv1.Network{}
+		if err := status.client.ClientFor("").CRClient().
+			Get(context.TODO(), types.NamespacedName{Name: names.OPERATOR_CONFIG}, netCfg); err == nil {
+			migrationInProgress = meta.IsStatusConditionPresentAndEqual(
+				netCfg.Status.Conditions,
+				names.NetworkTypeMigrationInProgress,
+				metav1.ConditionTrue,
+			)
+		}
+
+		if oc.Spec.DefaultNetwork.Type == operv1.NetworkTypeOpenShiftSDN && migrationInProgress {
```

Comment:
I think we shall remove the check `oc.Spec.DefaultNetwork.Type == operv1.NetworkTypeOpenShiftSDN`. Upgrade shall be blocked when `migrationInProgress` is true.

Reply:
done